packages/salesforce-adapter/src/change_validators/custom_applications.ts

```diff
@@ -42,59 +45,64 @@ const isValidActionOverride = (action: unknown): boolean =>
+    key.push(isReferenceExpression(action.recordType) ? action.recordType.elemID.getFullName() : action.recordType)
+  }
+  return key.join('')
 }
 
-const collectDuplicates = (actions: Array<ActionOverride | ProfileActionOverride>): Set<string> => {
-  const seen = new Set<string>()
-  const duplicates = new Set<string>()
-  actions.forEach(action => {
+const collectDuplicates = (
+  actions: Array<ActionOverride | ProfileActionOverride>,
+  fieldName: string,
+  instance: InstanceElement,
+): ElemID[] => {
+  const duplicates = new DefaultMap<string, ElemID[]>(() => [])
+  actions.forEach((action, index) => {
     const key = generateActionOverrideKey(action)
-    if (seen.has(key)) {
-      duplicates.add(key)
-    } else {
-      seen.add(key)
-    }
+    duplicates.get(key).push(instance.elemID.createNestedID(fieldName, index.toString()))
   })
-  return duplicates
+  return Array.from(duplicates.entries()).flatMap(([_key, elemIds]) => (elemIds.length > 1 ? elemIds : []))
 }
 
-const createChangeError = (duplicates: Set<string>, elemId: ElemID): ChangeError => {
-  const duplicateList = Array.from(duplicates)
-    .map(dup => `- ${dup}`)
-    .join('\n')
-  return {
+const createChangeError = (elemIds: ElemID[]): ChangeError[] =>
+  elemIds.map(elemId => ({
     elemID: elemId,
-    severity: 'Error',
-    message: 'Custom Application has conflicting action overrides',
-    detailedMessage: `The following action overrides have multiple definitions:\n${duplicateList}`,
-  }
-}
+    severity: 'Warning',
+    message: 'Duplicate action override',
+    detailedMessage: 'This action override has multiple definitions',
+  }))
 
-const findActionOverridesDuplications = (instance: InstanceElement): ChangeError | undefined => {
+const findActionOverridesDuplications = (instance: InstanceElement): ChangeError[] => {
   const values: unknown = instance.value
-  if (!isCustomApplicationWithOverrides(values)) {
-    return undefined
-  }
-  const actionOverridesArray = values.actionOverrides ?? []
-  const profileActionOverridesArray = values.profileActionOverrides ?? []
-  const allActions = actionOverridesArray.concat(profileActionOverridesArray)
-  const duplicates = collectDuplicates(allActions)
-  return duplicates.size > 0 ? createChangeError(duplicates, instance.elemID) : undefined
+  const actionOverridesArray = hasActionOverrides(values) ? values.actionOverrides : []
+  const profileActionOverridesArray = hasProfileActionOverrides(values) ? values.profileActionOverrides : []
+  const actionOverridesDuplicates = collectDuplicates(actionOverridesArray, 'actionOverrides', instance)
+  const profileActionOverridesDuplicates = collectDuplicates(
+    profileActionOverridesArray,
+    'profileActionOverrides',
+    instance,
+  )
+  const actionOverridesErrors = createChangeError(actionOverridesDuplicates)
+  const profileActionOverridesErrors = createChangeError(profileActionOverridesDuplicates)
```

Comment:
I would condense this a lot:
```suggestion
  const actionOverridesErrors = hasActionOverrides(values) ? createChangeError(collectDuplicates(values.actionOverrides, 'actionOverrides', instance)) : []
  const profileActionOverridesErrors = hasProfileActionOverrides(values) ? createChangeError(collectDuplicates(
    values.profileActionOverrides,
    'profileActionOverrides',
    instance,
  )) : []
```